File: script/tc_analysis.py
#### Packages ####
import pandas as pd
import glob, os
import re

#### Load data ####
os.chdir("/Users/pomato/mrc/project/sita_slam-seq/")

tc_dir = "./processed/PROJ1624_TC"
files = glob.glob(f"{tc_dir}/*.csv")

tc_rates = []

# Calculate T > C conversion rate per sample
for file in files:
    # Get sample name
    sample = os.path.basename(file).replace("_R1_trimmed.fq_slamdunk_mapped_filtered_mutationrates_utr.csv", "")
    
    # Format grouping
    group = re.sub(r'^T[0-9]_', '', re.sub(r'_rep[0-9]$', '', sample))

    # Calculate overall T > C conversion rate
    df = pd.read_csv(file, sep="\t", header=2)
    tc = df['T_C'].sum() / (df['T_C'].sum() + df['T_T'].sum() + df['T_A'].sum() + df['T_G'].sum()) * 100
    
    tc_rates.append([sample, group, tc])

tc_df = pd.DataFrame(tc_rates, columns=['sample', 'group', 'TC'])

# Save output
tc_df.to_csv("./processed/PROJ1624_TC.txt", sep="\t", index=False)

File: script/tc_analysis.R
setwd("/Users/pomato/mrc/project/sita_slam-seq/")

#### Packages ####
require(ggplot2)

#### Load data ####
tc <- read.table("./processed/PROJ1791_overallTCrates.txt",
                 header=TRUE, sep="\t", check.names=FALSE)

#### Visualise data ####
plot <- ggplot(data=tc, aes(x=group, y=TC, color=factor(rep))) +
  geom_point(position=position_dodge(width=0.7)) +
  scale_colour_manual(values=rep("black", 5)) +
  stat_summary(fun.data="mean_cl_boot", geom="errorbar",width=0.05, colour="red") +
  stat_summary(fun=mean, fun.max=mean, fun.min=mean, colour="red", geom="crossbar", width=0.2) +
  theme(legend.position="none") +
  xlab("") +
  ylab("Overall % T>C conversion rate")

png("./processed/PROJ1791_overallTCrates.png")
plot
dev.off()